pkg/analyzer/nodes/node.go

```diff
@@ -103,22 +104,51 @@ func extractFilesFromDirectory(root string, matcher matcher.PrefixMatcher) (*fil
 	m := metrics.FileExtractionMetrics{}
 	// TODO(ROX-13771): Use `range matcher.GetCommonPrefixDirs()` again after fixing.
 	for _, dir := range []string{"etc/", "usr/share/rpm", "var/lib/rpm", "usr/share/buildinfo"} {
-		if err := n.addFiles(filepath.FromSlash(dir), matcher, &m); err != nil {
-			return nil, errors.Wrapf(err, "failed to match filesMap at %q (at %q)", dir, n.root)
+		if err := n.addFiles(ctx, filepath.FromSlash(dir), matcher, &m); err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				err = errors.Wrapf(err, "operation took longer than expected and was interrupted")
```

Comment:
I'm not sure this really adds anything. The error message is already "context deadline exceeded" https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/context/context.go;l=171

Reply:
Doesn't it give readers additional context on what precisely happened? In particular, that the file extraction was interrupted because it took too long to extract the files from a specific directory. Does "context deadline exceeded" convey the same information?

Reply:
I think "context deadline exceeded" already says the same thing as "operation took longer than expected and was interrupted". If this is to be used for logging purposes, then perhaps the caller can say something like: "Unable to complete analysis: <error>" or even customize the log if `errors.Is(err context.DeadlineExceeded)` since the caller knows this is a context with a deadline and this is a possible error